crates/forge-runner/src/coverage_api.rs

```diff
@@ -0,0 +1,42 @@
+use anyhow::{Context, Result};
+use shared::command::CommandExt;
+use std::process::Stdio;
+use std::{env, fs, path::PathBuf, process::Command};
+
+pub const COVERAGE_DIR: &str = "coverage";
+pub const OUTPUT_FILE_NAME: &str = "coverage.lcov";
+
+pub fn run_coverage(saved_trace_data_paths: &[PathBuf]) -> Result<()> {
+    let coverage = env::var("CAIRO_COVERAGE")
+        .map(PathBuf::from)
+        .ok()
+        .unwrap_or_else(|| PathBuf::from("cairo-coverage"));
+
+    let dir_to_save_coverage = PathBuf::from(COVERAGE_DIR);
+    fs::create_dir_all(&dir_to_save_coverage).context("Failed to create a coverage dir")?;
+    let path_to_save_coverage = dir_to_save_coverage.join(OUTPUT_FILE_NAME);
+
+    let trace_files: Vec<&str> = saved_trace_data_paths
+        .iter()
+        .map(|trace_data_path| {
+            trace_data_path
+                .as_path()
+                .to_str()
+                .expect("Failed to convert trace data path to string")
```

Comment:
Why do we panic here instead of returning and `Err`?

Reply:
A bit easier this way, and given that it should never fail I think it is acceptable to use `expect` here